### PR TITLE
Fix testnet and optimize performance

### PR DESCRIPTION
**Pull request by maraoz**

### CODE REVIEW

File: lib/networkmonitor.js [hunk 1]
```diff
@@ -9,13 +9,16 @@ var $ = bitcore.util.preconditions;
...
-var messages = new p2p.Messages();
 
 function NetworkMonitor(eventBus, peer) {
   $.checkArgument(eventBus);
   $.checkArgument(peer);
   this.bus = eventBus;
   this.peer = peer;
+  this.messages = new p2p.Messages({
+    network: this.peer.network,
+    magicNumber: this.peer.network.networkMagic.readUInt32LE(0)
```
Comment (braydonf) [inline lib/networkmonitor.js]: magicNumber isn't needed anymore with update to bitcore-p2p v0.13.0

File: lib/node.js [hunk 2]
```diff
@@ -82,8 +85,10 @@ BitcoreNode.prototype.initialize = function() {
     }
     var tipHash = self.blockchain.tip;
     var block = self.blockCache[tipHash];
-    console.log('block', block.id, 'height', block.height);
-  }, 5 * 1000);
+    var delta = block.height - prevHeight;
+    prevHeight = block.height;
+    console.log(block.id, block.height, 'vel', delta * 1000 / statTimer, 'b/s');
```
Comment (braydonf) [inline lib/node.js]: There is a lib/logger.js should that be used instead of console.log?

Comment (maraoz) [inline lib/node.js]: That's updated. I'll improve logging in a future PR

File: package.json [hunk 3]
```diff
@@ -6,28 +6,16 @@
...
     },
     {
-      "name": "Juan Ignacio Sosa Lopez",
-      "email": "bechilandia@gmail.com"
-    },
-    {
-      "name": "Ivan Socolsky",
-      "email": "jungans@gmail.com"
+      "name": "Manuel Araoz",
+      "email": "manuelaraoz@gmail.com"
```
Comment (braydonf) [inline package.json]: Doubled

File: lib/services/transaction.js [hunk 4]
```diff
@@ -50,15 +50,15 @@ var helperAddress = function(index) {
...
+  output: 'txo-', // txo-<txid>-<n> -> serialized Output
+  spent: 'txs-', // txo-<txid>-<n>-<spend txid>-<m> -> block height of confirmation for spend
+  address: 'txa-', // txa-<address>-<txid>-<n> -> Output
   addressSpent: 'txas-', // txa-<address>-<txid>-<n> -> {
-                         //   heightSpent: number,          (may be -1 for unconfirmed tx)
-                         //   spentTx: string, spentTxInputIndex: number, spendInput: Input
-                         // }
-  transaction: 'btx-'   // btx-<txid> -> block in main chain that confirmed the tx
-}
+  //   heightSpent: number,          (may be -1 for unconfirmed tx)
```
Comment (braydonf) [inline lib/services/transaction.js]: Intentionally commented?

File: lib/services/block.js [hunk 5]
```diff
@@ -251,6 +253,7 @@ BlockService.prototype.confirm = function(block, ops) {
       //console.log(4);
       self._setBlockByTs(ops, block);
 
+      //console.log(4.1);
```
Comment (braydonf) [inline lib/services/block.js]: Debug logs?

Comment (maraoz) [inline lib/services/block.js]: Yes, I'll keep those for now, it's helpful for optimizations (will remove before release)

File: lib/services/transaction.js (hunk 4 above)
Comment (maraoz) [inline lib/services/transaction.js]: only formatting changed, will fix for clarity

**Pull request merged**